### ISSUE TITLE
Create a wrapper around `Vec` that supports indexing by `u32`

### ISSUE THREAD

**Issue by akoshelev** [martinthomson/raw-ipa#116]
see the discussion here: https://github.com/martinthomson/raw-ipa/pull/108#discussion_r992621416

It would be great to have a type inside IPA repo that defines upper bound for IPA input size. We target to be able to handle 1B events, so a typealias (`RecordIndex`/`RecordId`?) for `u32` is enough for now. It probably makes sense to have a vector type that allows indexing by this type without casting to usize and back - that should remove a lot of useless conversions from the protocol code. 

tldr

```rust
   typealias RecordIndex = u32;
   struct RecordsVec<T> {
       inner: Vec<T>
   }

   impl TryFrom<Vec<T>> for RecordsVec<T> { ... }
   impl <T> Index<u32> for RecordsVec<T> { ... }
   impl AsRef<[T]> for RecordsVec<T> { ... }
   impl AsMut<[T]> for RecordsVec<T> { ... }
   impl Borrow<[T]> for RecordsVec<T> { ... }
   impl BorrowMut<[T]> for RecordsVec<T> { ... }
   impl <T> IntoIterator for RecordsVec<T> { ... }
```
